synapse/storage/databases/main/__init__.py

```diff
@@ -203,6 +203,7 @@ async def get_users_paginate(
         deactivated: bool = False,
         order_by: str = UserSortOrder.USER_ID.value,
         direction: str = "f",
+        approved: bool = True,
```

Comment:
I find this arg name, both here in the method and in the admin API, non-obvious: it sounds like I ought to be able to write `approved=true` to get only approveds and `approved=false` to get only un-approveds.
I would be tempted to make it `Optional[bool]` so you get the choice of unfiltered or filtered either way.
Alternatively, to cater only to the 'get list of users needing approval' use case, I'd probably call it `needs_approval` and invert the meaning.

Reply:
Good point, though the reason I'm doing it like this is to be consistent with other parameters such as `guests`. I think it's clearer that way because similar parameters imply that if they're absent from the request then the category of users they refer to will be included in the response (whereas `needs_approval` kind of implies otherwise). But then we also have the `deactivated` parameter which works kind of the same way but with flipped values of true/false and aaargh why is life so terrible.
But in general I still think it makes more sense to have an `approved` parameter.

Reply:
I see the dilemma here for either doing it 'consistently silly' or being inconsistent but maybe more useful.

I think I'd still prefer to do it inconsistently but less sillily, but don't feel too strongly about it so if you're not convinced that's fine by me.

Reply:
I do agree that it would be worth changing this API to fix this because it's confusing, but I'd rather we do it once in a consistent way (in a separate PR) rather than introduce inconsistencies that could confuse the users even more.